File: Module1/casestudy/yeucau6/task_2/Employee.js
let Employee = function () {
    this.getName = function () {
        return this.name;
    };
    this.getBirthday = function () {
        return this.birthday;
    };
    this.getPhone = function () {
        return this.phone;
    };
    this.getEmail = function () {
        return this.email;
    };
    this.getCmnd = function () {
        return this.cmnd;
    };
    this.getLevel = function () {
        return this.level;
    };
    this.getLocation = function () {
        return this.location;
    };
    this.getSalary = function () {
        if (this.getLocation()=== "Manager"){
            this.salary = 500;
        }else if (this.getLocation()=== "Sale"){
            this.salary = 300;
        }else if (this.getLocation()==="Makerting"){
            this.salary = 200;
        }
        return this.salary;

    };
    this.setName = function (name) {
        this.name = name;
    };
    this.setBirthday = function (birthday) {
        this.birthday = birthday;
    };
    this.setPhone = function (phone) {
        this.phone = phone;
    };
    this.setEmail = function (email) {
        this.email = email;
    };
    this.setCmnd = function (cmnd) {
        this.cmnd = cmnd;
    };
    this.setLevel = function (level) {
        this.level = level;
    };
    this.setLocation = function (location) {
        this.location = location;
    };
    this.setSalary = function (salary) {
        this.salary = salary;
    };

};

File: Module4/20_case_study/case_study/src/main/java/vn/codegym/model/Employee.java
package vn.codegym.model;

import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.Valid;
import javax.validation.constraints.*;
import java.time.LocalDate;
import java.util.Set;

@Entity
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @NotBlank(message = "Không được để trống")
    private String name;

    @NotNull(message = "Không được để trống")
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    @Past(message = "Chỉ nhập ngày trong quá khứ")
    private LocalDate birthday;

    @Pattern(regexp = "^(\\d{9})$", message = "CMND gồm 9 số")
    private String idCard;

    @Min(value = 0, message = "Tối thiểu là 0 VND")
    private double salary;

    @Pattern(regexp = "^09[0-1][0-9]{7}$", message = "Phải đúng định dạng 090xxxxxxx ")
    private String phone;

    @Pattern(regexp = "^[a-z][a-z0-9_\\.]{5,32}@[a-z0-9]{2,}(\\.[a-z0-9]{2,4}){1,2}$",
            message = "Phải đúng định dạng abc@xyz.com")
    private String email;

    @NotBlank(message = "Không được để trống")
    private String address;

    @ManyToOne
    private Position position;

    @ManyToOne
    private EducationDegree educationDegree;

    @ManyToOne
    private Department department;

    @OneToOne(cascade = CascadeType.ALL)
    @Valid
    private AppUser appUser;

    @OneToMany(mappedBy = "employee")
    private Set<Contract> contracts;

    public Employee() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LocalDate getBirthday() {
        return birthday;
    }

    public void setBirthday(LocalDate birthday) {
        this.birthday = birthday;
    }

    public String getIdCard() {
        return idCard;
    }

    public void setIdCard(String idCard) {
        this.idCard = idCard;
    }

    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public EducationDegree getEducationDegree() {
        return educationDegree;
    }

    public void setEducationDegree(EducationDegree educationDegree) {
        this.educationDegree = educationDegree;
    }

    public Department getDepartment() {
        return department;
    }

    public void setDepartment(Department department) {
        this.department = department;
    }

    public AppUser getAppUser() {
        return appUser;
    }

    public void setAppUser(AppUser appUser) {
        this.appUser = appUser;
    }

    public Set<Contract> getContracts() {
        return contracts;
    }

    public void setContracts(Set<Contract> contracts) {
        this.contracts = contracts;
    }


    public Position getPosition() {
        return position;
    }

    public void setPosition(Position position) {
        this.position = position;
    }
}
